src/commands/setup.rs

```diff
@@ -69,10 +70,9 @@ impl Setup {
                 "bridge" => {
                     let per_network_opts =
                         network_options.networks.get(net_name).ok_or_else(|| {
-                            std::io::Error::new(
-                                std::io::ErrorKind::Other,
-                                format!("network options for network {} not found", net_name),
-                            )
+                            anyhow!(NetavarkErrorCode::ErrNoNetworkOptions {
```

Comment:
Is there any specific case for `anyhow!`. I am still not sure but I would also request others to pitch in since i was ok with wrapping in the first place.

something like 
```rust
return NetavarkError::Error(err, "my custom wrap message".to_string())
```

Reply:
I thought this again  I think we just need a new function something like

```rust
return NetavarkError::Error(err, "my custom wrap message".to_string(), ERR_CODE);
```
and NetavarkError::Error returns a valid `NetavarkError` 

so we could remove
```rust
std::io::Error::new( std::io::ErrorKind::Other,...
```

@baude @Luap99 @mheon please correct me if i am wrong.

Reply:
You're correct, but I would be fine with using an existing library for wrapping errors (I imagine one does exist already) instead of implementing this ourselves. I don't actually see `anyhow!()` wrapping, though, it accepts an error but not a string to annotate said error with?

Reply:
@mheon @flouthoc Do we need separate annotations if it's mentioned in the enum itself along with the error code? I can update this to get rid of the anyhow!() and use a function to return error code + custom msg if rqeuired. Do you see any issues with the `thiserror` enum created? 

A bunch of crates like `error_chain` and `failure` are now deprecated. anyhow is recommended now instead of failure and thiserror is recommended instead of error_chain.

Let me know.